control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -489,16 +500,58 @@ func (r Reconciler) reconcileKedaObjects(ctx context.Context, cg *kafkainternals
-		if cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef != nil {
-			secretKeyRefName := cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef.Name
-			secretKeyRefKey := cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef.Key
-			secret, err := r.KubeClient.CoreV1().Secrets(cg.Namespace).Get(ctx, secretKeyRefName, metav1.GetOptions{})
-			if err != nil {
-				return nil, reconciler.NewEvent(corev1.EventTypeWarning, "SaslTypeSecretUnavailable", "Unable to get SASL type from secret: \"%s/%s\", %w", cg.Namespace, secretKeyRefName, err)
-			}
-			saslTypeValue := string(secret.Data[secretKeyRefKey])
-			return &saslTypeValue, nil
+	if hasNetSpecAuthConfig(cg.Spec.Template.Spec.Auth) && cg.Spec.Template.Spec.Auth.NetSpec.SASL.Enable && cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef != nil {
+		secretKeyRefName := cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef.Name
+		secretKeyRefKey := cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef.Key
+		secret, err := r.KubeClient.CoreV1().Secrets(cg.Namespace).Get(ctx, secretKeyRefName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to get SASL type from secret: \"%s/%s\", %w", cg.Namespace, secretKeyRefName, err)
+		}
+		saslTypeValue := string(secret.Data[secretKeyRefKey])
+		return &saslTypeValue, nil
+	}
+	if hasSecretSpecConfig(cg.Spec.Template.Spec.Auth) && cg.Spec.Template.Spec.Auth.SecretSpec.Ref != nil {
+		secretKeyRefName := cg.Spec.Template.Spec.Auth.SecretSpec.Ref.Name
+		secretKeyRefNamespace := cg.Spec.Template.Spec.Auth.SecretSpec.Ref.Namespace
+		secret, err := r.KubeClient.CoreV1().Secrets(secretKeyRefNamespace).Get(ctx, secretKeyRefName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to get SASL type from secret: \"%s/%s\", %w", secretKeyRefNamespace, secretKeyRefName, err)
+		}
+		if saslTypeValue, ok := secret.Data[security.SaslType]; ok {
+			saslType := string(saslTypeValue)
+			return &saslType, nil
+		}
+	}
+	return nil, nil
+}
+
+func (r *Reconciler) retrieveProtocolIfPresent(ctx context.Context, cg *kafkainternals.ConsumerGroup) (*string, error) {
+	if hasSecretSpecConfig(cg.Spec.Template.Spec.Auth) && cg.Spec.Template.Spec.Auth.SecretSpec.Ref != nil {
+		secretKeyRefName := cg.Spec.Template.Spec.Auth.SecretSpec.Ref.Name
+		secretKeyRefNamespace := cg.Spec.Template.Spec.Auth.SecretSpec.Ref.Namespace
+		secret, err := r.KubeClient.CoreV1().Secrets(secretKeyRefNamespace).Get(ctx, secretKeyRefName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to get Protocol from secret: \"%s/%s\", %w", secretKeyRefNamespace, secretKeyRefName, err)
+		}
+		if protocolValue, ok := secret.Data[security.ProtocolKey]; ok {
+			protocol := string(protocolValue)
+			return &protocol, nil
+		}
+	}
+	return nil, nil
+}
+
+func (r *Reconciler) retrieveTlsCertAuthIfPresent(ctx context.Context, cg *kafkainternals.ConsumerGroup) (*string, error) {
```

Comment:
why "if present" ?

Reply:
For SASL secrets (no SSL), the certificate authority field won't be present. Hence the `if`. We're using its existence to determine if TLS must be enabled for KEDA auth. Hope that helps.